usecases/schema/schema_repair.go

```diff
@@ -0,0 +1,157 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package schema
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/schema"
+	"github.com/weaviate/weaviate/usecases/sharding"
+)
+
+type clusterSyncRepairs struct {
+	classes []*models.Class
+	props   map[string][]*models.Property
+	tenants map[string][]sharding.Physical
+}
+
+func newClusterSyncRepairs() clusterSyncRepairs {
+	return clusterSyncRepairs{
+		props:   make(map[string][]*models.Property),
+		tenants: make(map[string][]sharding.Physical),
+	}
+}
+
+type comparisonState struct {
+	*State
+	classSet map[string]*models.Class
+}
+
+func newComparisonState(state *State) comparisonState {
+	return comparisonState{
+		State:    state,
+		classSet: classSliceToMap(state.ObjectSchema.Classes),
+	}
+}
+
+type comparisonUnit struct {
+	anomaly, comparator comparisonState
+}
+
+func (m *Manager) repairSchema(ctx context.Context, remote *State) error {
+	local := &m.schemaCache.State
+	if local.ObjectSchema == nil || remote.ObjectSchema == nil {
+		return fmt.Errorf("nil schema found, local: %v, remote: %v",
+			local.ObjectSchema, remote.ObjectSchema)
+	}
+
+	local2Remote := comparisonUnit{
+		anomaly:    newComparisonState(local),
+		comparator: newComparisonState(remote),
+	}
+	localRepairs := determineRepairsNeeded(local2Remote)
+
+	// TODO: repair remote node schemas as well
+	// remote2Local := comparisonUnit{
+	// 	 anomaly:    newComparisonState(remote),
+	//	 comparator: newComparisonState(local),
+	// }
+	// remoteRepairs := determineRepairsNeeded(remote2Local)
+
+	for _, class := range localRepairs.classes {
+		shardState, err := sharding.InitState(class.Class,
+			class.ShardingConfig.(sharding.Config),
+			m.clusterState, class.ReplicationConfig.Factor,
+			schema.MultiTenancyEnabled(class))
+		if err != nil {
+			return fmt.Errorf("init sharding state: %w", err)
+		}
+		if err = m.addClassApplyChanges(ctx, class, shardState); err != nil {
+			return fmt.Errorf("add class %q locally: %w", class.Class, err)
+		}
+	}
+
+	for class, props := range localRepairs.props {
+		for _, prop := range props {
+			_, err := m.schemaCache.addProperty(class, prop)
```

Comment:
Shouldn't we also add the property to the index, similar to how it's done for classes?

Reply:
Hey, actually we don't need to do this, because repair happens before the DB is initiated. 

The DB uses the schema as the source of truth, and inits all resources based off that. So when the property buckets are initiated, any previously missing ones are created.